org.eclipse.winery.repository.ui/src/app/instance/instanceHeader/propertyRename/propertyRename.service.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2017 University of Stuttgart.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and the Apache License 2.0 which both accompany this distribution,
+ * and are available at http://www.eclipse.org/legal/epl-v10.html
+ * and http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Contributors:
+ *     Lukas Balzer - initial API and implementation
+ */
+import { Injectable } from '@angular/core';
+import { Headers, Http, RequestOptions, Response } from '@angular/http';
+import { Router } from '@angular/router';
+import { backendBaseURL } from '../../../configuration';
+import { ToscaComponent } from '../../../wineryInterfaces/toscaComponent';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class PropertyRenameService {
+
+    private url: string;
+    private id: string;
+    private resourceType: string;
+    private propertyName: string;
+    private toscaComponent: ToscaComponent;
+
+    constructor(private http: Http,
+                private route: Router) {
+    }
+
+    setId(id: string) {
+        this.id = id;
+    }
+
+    setPropertyName(name: string) {
+        this.propertyName = name;
+    }
+
+    setToscaComponent(toscaComponent: ToscaComponent) {
+        this.toscaComponent = toscaComponent;
+        this.resourceType = decodeURIComponent(this.route.url.split('/')[0]
```

Comment:
The `toscaComponent` also has a `toscaType`. This is a special enum which decodes all our resources. Because it's a string enum, there is no need for getting the resource type this way. Just use `toscaComponent.toscaType`.

The `toscaComponent` also includes the path to the current component. No need for generation